pennylane/templates/subroutines/approx_time_evolution.py

```
@@ -0,0 +1,147 @@
+           time-evolution operator. The indices of the observables in the Hamiltonian
+           correspond to the ``wires`` element to which each observable is being applied.
+           The Hamiltonian must be explicitly written
+           in terms of products of Pauli gates (X, Y, Z, and I).
+
+        time (int or float): The time of evolution, namely the parameter :math:`t` in :math:`e^{- i H t}`.
+
+        n (int): The number of Trotter steps used when approximating the time-evolution operator.
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        The template is used inside a qnode:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import ApproxTimeEvolution
+
+            n_wires = 2
+            wires = range(n_wires)
+
+            dev = qml.device('default.qubit', wires=n_wires)
+
+            coeffs = [1, 1]
+            obs = [qml.PauliX(0), qml.PauliX(1)]
+            hamiltonian = qml.Hamiltonian(coeffs, obs)
+
+            @qml.qnode(dev)
+            def circuit(time):
+                ApproxTimeEvolution(hamiltonian, time, 1)
+                return [qml.expval(qml.PauliZ(wires=i)) for i in wires]
+
+        >>> circuit(1)
+        [-0.41614684 -0.41614684]
+    """
+
+    pauli = {"Identity": "I", "PauliX": "X", "PauliY": "Y", "PauliZ": "Z"}
+
+    if not isinstance(hamiltonian, qml.vqe.vqe.Hamiltonian):
+        raise ValueError(
+            "`hamiltonian` must be of type pennylane.Hamiltonian, got {}".format(
+                type(hamiltonian).__name__
+            )
+        )
+
+    if not isinstance(time, (int, float, qml.variable.Variable)):
+        raise ValueError("`time` must be of type int or float, got {}".format(type(time).__name__))
+
+    if not isinstance(n, (int, qml.variable.Variable)):
+        raise ValueError("`n` must be of type int, got {}".format(type(n).__name__))
+
```

```suggestion
    if not isinstance(time, (int, float, qml.variable.Variable)):
        raise ValueError("time must be of type int or float, got {}".format(type(time).__name__))

    if not isinstance(n, (int, qml.variable.Variable)):
        raise ValueError("n must be of type int, got {}".format(type(n).__name__))

```

---

Might be good to double check with @mariaschuld, I'm not sure if we are being so strict on these input validations any more. In particular, `t` could be a TensorFlow tensor or an Autograd ArrayBox if doing backpropagation.

---

@co9olguy @mariaschuld Did you have any thoughts on this before we merge?

---

@josh146 Yeah, that makes sense. If we change the input validations to account for tensors and Autograd ArrayBoxes, would this issue be resolved?

---

I think we should just take this test out all together and let errors be thrown when the circuit is being built, if the user tries to input something weird